File: src/TrayAppRUI/ViewModels/MainWindowViewModel.fs
namespace FSharpReactiveUI.ViewModels

open System.Reflection

type MainWindowViewModel(queryFunc: string -> Result<string, string>) =
    inherit ViewModelBase()

    let version =
        Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>()
        |> Option.ofObj
        |> Option.map (fun a -> a.InformationalVersion.Split('+')[0])
        |> Option.defaultValue "0.0-error"

    member val QueryPanel = QueryPanelViewModel(queryFunc)
    member _.VersionDisplay = $"Version {version}"

File: src/TrayAppCS/ViewModels/MainWindowViewModel.cs
using System;
using System.Reflection;
using Microsoft.FSharp.Core;
using ReactiveUI;

namespace GlazeWM.TrayAppCS.ViewModels;

public class MainWindowViewModel : ReactiveObject
{
    public QueryPanelViewModel QueryPanel { get; }

    public string VersionDisplay { get; }

    public MainWindowViewModel(Func<string, FSharpResult<string, string>> queryFunc)
    {
        QueryPanel = new QueryPanelViewModel(queryFunc);

        var version = Assembly.GetExecutingAssembly()
            .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
            ?.InformationalVersion.Split('+')[0] ?? "0.0-error";

        VersionDisplay = $"Version {version}";
    }
}
